pkg/checkup/status.go

```diff
@@ -0,0 +1,116 @@
+package checkup
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	statusSucceededKey        = "status.succeeded"
+	statusFailureReasonKey    = "status.failureReason"
+	statusStartTimestamp      = "status.startTimestamp"
+	statusCompletionTimestamp = "status.completionTimestamp"
+	statusResultNameKeyPrefix = "status.result."
+)
+
+type Status struct {
+	succeeded           string
+	failureReason       string
+	startTimestamp      string
+	completionTimestamp string
+	results             map[string]string
+}
+
+func NewStatusFromConfigMap(configMap *corev1.ConfigMap) (*Status, error) {
+	status := &Status{}
+
+	if configMap.Data == nil {
+		return nil, fmt.Errorf("data field is missing from ConfigMap")
+	}
+
+	var exists bool
+
+	status.succeeded, exists = configMap.Data[statusSucceededKey]
+	if !exists {
+		return nil, fmt.Errorf("failed to read succeeded field")
+	}
```

Comment:
The status properties should be added by the framework, if they are missing they should be added instead of failing.